contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,843 @@
+        if (balance > 0) {
+            _withdraw(vaultAddress, balance);
+        }
+        // burn remaining OETHb
+        _burnOETHbOnTheContract();
+    }
+
+    function _withdraw(
+        address _recipient,
+        uint256 _amount
+    ) internal {
+        require(_amount > 0, "Must withdraw something");
+        require(_recipient != address(0), "Must specify recipient");
+
+        IERC20(WETH).safeTransfer(_recipient, _amount);
+        emit Withdrawal(WETH, address(0), _amount);
+    }
+
+
+    /**
+     * @dev Collect the AERO token from the gauge 
+     */
+    function _collectRewardTokens() internal override {
+        // TODO do other stuff
+        super._collectRewardTokens();
+    }
+
+    /**
+     * @dev Retuns bool indicating whether asset is supported by strategy
+     * @param _asset Address of the asset
+     */
+    function supportsAsset(address _asset) public view override returns (bool) {
+        return _asset == WETH;
+    }
+
+    /**
+     * @dev Internal method to respond to the addition of new asset / pTokens
+            We need to give the Aerodrome pool approval to transfer the
+            asset.
+     * @param _asset Address of the asset to approve
+     * @param _pToken Address of the pToken
+     */
+    // solhint-disable-next-line no-unused-vars
+    function _abstractSetPToken(address _asset, address _pToken)
+        internal
+        override
+    {
+        IERC20(_asset).safeApprove(address(positionManager), type(uint256).max);
+        IERC20(_asset).safeApprove(address(swapRouter), type(uint256).max);
+    }
+
+    /**
+     * @dev Approve the spending of all assets 
+     */
+    function safeApproveAllTokens()
+        external
+        override
+        onlyGovernor
+        nonReentrant
+    {
+        IERC20(WETH).safeApprove(address(positionManager), type(uint256).max);
+        IERC20(OETHb).safeApprove(address(positionManager), type(uint256).max);
+        IERC20(WETH).safeApprove(address(swapRouter), type(uint256).max);
+        IERC20(OETHb).safeApprove(address(swapRouter), type(uint256).max);
+    }
+
+    /***************************************
+            Balances and Fees
+    ****************************************/
+
+    /**
+     * @dev Get the total asset value held in the platform
+     * @param _asset      Address of the asset
+     * @return balance    Total value of the asset in the platform
+     */
+    function checkBalance(address _asset)
+        external
+        view
+        override
+        returns (uint256)
+    {
+        return netValue;
+    }
+
+    /**
+     * @dev Returns the balance of both tokens in a given position (excluding fees)
+     * @return amountWETH Amount of WETH in position
+     * @return amountOETHb Amount of OETHb in position
+     */
+    function getPositionPrincipal()
+        public
+        view
+        returns (uint256 amountWETH, uint256 amountOETHb)
+    {
+        if (tokenId == 0) {
+            revert("No LP position");
```

Comment:
Why revert here instead of returning 0,0? This, for example, breaks withdrawAll. If we want to have things revert on no tokenId, that's probably better handled by a separate function with a name that makes it clear what it's doing.